newIDE/app/src/Hints/ObjectsAdditionalWork.js

```diff
@@ -0,0 +1,121 @@
+// @flow
+import { t } from '@lingui/macro';
+import { type AlertMessageIdentifier } from '../MainFrame/Preferences/PreferencesContext';
+import newNameGenerator from '../Utils/NewNameGenerator';
+
+/*
+ * Define additional logic which executes after an object/instance has been created.
+ * Also, InfoBar can be used which could notify users of the additional changes.
+ * Declare new identifier for infoBar in Mainframe/Preferences/PreferenceContext
+ * and add it in hints/explaination list.
+ */
+
+export type InfoBarDetails = {
+  show: boolean,
+  identifier: AlertMessageIdentifier,
+  message: string,
+  touchScreenMessage: string,
+};
+
+type InfoBarEvent = 'onObjectAdded' | 'onInstanceAdded';
+
+export default {
+  onObjectAdded(
```

Comment:
nit: Consider using `export const onObjectAdded = ...` - this favors the use of functions rather than an object, so it helps static analysis and in the future can help bundlers like Rollup to do "Tree shaking". Seems like named exports (`export const ...`/`export class ...`) are becoming preferred to default exports for these reasons in the JS community